vsintegration/src/FSharp.Editor/CodeFix/RenameParamToMatchSignature.fs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Collections.Immutable
+open System.Collections.Generic
+open System.Composition
+open System.Threading
+open System.Threading.Tasks
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+open Microsoft.CodeAnalysis.CodeActions
+
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.Range
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open SymbolHelpers
+
+[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = "FSharpRenameParamToMatchSignature"); Shared>]
+type internal FSharpRenameParamToMatchSignature
+    [<ImportingConstructor>]
+    (
+        checkerProvider: FSharpCheckerProvider, 
+        projectInfoManager: FSharpProjectOptionsManager
+    ) =
+    
+    inherit CodeFixProvider()
+    static let userOpName = "RenameParamToMatchSignature"
+    let fixableDiagnosticIds = ["FS3218"]
+        
+        
+    override __.FixableDiagnosticIds = Seq.toImmutableArray fixableDiagnosticIds
+
+    override __.RegisterCodeFixesAsync context : Task =
+        asyncMaybe {
+            match context.Diagnostics |> Seq.filter (fun x -> fixableDiagnosticIds |> List.contains x.Id) |> Seq.toList with 
+            | [diagnostic] -> 
+                    let message = diagnostic.GetMessage()
+                    let parts = System.Text.RegularExpressions.Regex.Match(message, ".*'(.*)'.*'(.*)'")
+                    if parts.Success then
+                    
+                        let diagnostics = ImmutableArray.Create diagnostic
+                        let suggestion = parts.Groups.[1].Value
+                        let replacement = if suggestion.Contains " " then "``" + suggestion + "``" else suggestion
```

Comment:
Can we not use this logic? I saw this used somewhere else and this isn't correct - there's many cases when you want to use \`\`

Reply:
@saul Yes, I agree.  I will find the right compiler helpe